pkg/ottl/contexts/ottlspan/span.go

```diff
@@ -55,7 +55,12 @@ func (tCtx TransformContext) GetResource() pcommon.Resource {
 }
 
 func NewParser(functions map[string]interface{}, telemetrySettings component.TelemetrySettings) ottl.Parser[TransformContext] {
```

Comment:
Should additional parameters after the functions parameter in each context's `NewParser` function also be optional? One benefit of using options is to prevent breaking changes when adding features to the parser,  and it seems more likely an OTTL consumer will be creating parsers for existing OTTL contexts than calling `ottl.NewParser` directly from their own custom context.

Reply:
Yes all the ottlcontext NewParser functions should allow options too.

Reply:
Shall each ottlcontext module have its own `Option` too?

Reply:
Currently they are used to obfuscate the generic types in the NewParser call. It's more future proof for them to have their own Option but I'd rather them continue to be pass through so I'd be good with using OTTL's Option.